finetrainers/trainer.py

```diff
@@ -479,30 +514,39 @@ def prepare_optimizer(self) -> None:
         params_to_optimize = [transformer_parameters_with_lr]
         self.state.num_trainable_parameters = sum(p.numel() for p in transformer_lora_parameters)
 
-        # TODO(aryan): add deepspeed support
+        use_deepspeed_opt = (
+            self.state.accelerator.state.deepspeed_plugin is not None 
+            and "optimizer" in self.state.accelerator.state.deepspeed_plugin.deepspeed_config
+        )
         optimizer = get_optimizer(
             params_to_optimize=params_to_optimize,
             optimizer_name=self.args.optimizer,
-            learning_rate=self.args.lr,
+            learning_rate=self.state.learning_rate,
             beta1=self.args.beta1,
             beta2=self.args.beta2,
             beta3=self.args.beta3,
             epsilon=self.args.epsilon,
             weight_decay=self.args.weight_decay,
+            use_deepspeed=use_deepspeed_opt
         )
 
         num_update_steps_per_epoch = math.ceil(len(self.dataloader) / self.args.gradient_accumulation_steps)
         if self.state.train_steps is None:
             self.state.train_steps = self.state.train_epochs * num_update_steps_per_epoch
             self.state.overwrote_max_train_steps = True
 
+        use_deepspeed_lr_schd = (
```

Comment:
```suggestion
        use_deepspeed_lr_scheduler = (
```

let's just be verbose for inline variables like this (does not have to be updated now, and we can tackle at once in future PR)